pipeline_dp/budget_accounting.py

```diff
@@ -113,10 +114,10 @@ def compute_budgets(self):
 class MechanismSpec:
     """Specifies the parameters for a mechanism.
 
-    NoiseKind defines the kind of noise distribution.
+    MechanismType defines the kind of noise distribution.
     noise is the minimized noise standard deviation.
     """
-    noise_kind: NoiseKind
+    noise_kind: MechanismType
```

Comment:
@dvadym I am not sure I understand exactly what to change here. I thought epsilon is always required, so I am missing something 🤔 Should I look more closely at the client API discussions? Thank you!

Reply:
`noise_standard_deviation` makes sense for Laplace/Gaussian, i.e. for noise addition mechanisms, but for generic (eps, delta) mechanism it is not applied, since we don't know anything about how this mechanism works. The only thing that we can use such mechism is (eps, delta) parameters, that what was my comment about.

So we can set `noise_standard_deviation` for Laplace/Gaussian and
set `(eps, delta)` for generic (eps, delta) mechanisms.

Reply:
Thanks for the clarification, that makes sense. I will make the changes 👍

Reply:
@dvadym All merged (`git reflog` is my friend 🤝), added epsilon and delta per generic mechanism along with tests. Thank you for 👀 and explanations!